sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluate/_evaluate.py

```diff
@@ -287,7 +287,13 @@ def _validate_columns_for_evaluators(
                 # Ignore the missing fields if "conversation" presents in the input data
                 missing_inputs = []
             else:
-                missing_inputs = [col for col in evaluator_params if col not in new_df.columns]
+                optional_params = (
+                    evaluator._OPTIONAL_PARAMS  # pylint: disable=protected-access
+                    if hasattr(evaluator, "_OPTIONAL_PARAMS")
+                    else []
+                )
+                excluded_params = set(new_df.columns).union(optional_params)
+                missing_inputs = [col for col in evaluator_params if col not in excluded_params]
```

Comment:
is there any test which covers this code?

Reply:
I added 3 tests in this PR:

1. Tests that you can run evaluate() with GroundednessEvaluator without the optional input in columns (and no error triggered from this section)
2. Tests that you can run evaluate() with GroundednessEvaluator with the optional input in columns
3. Tests that you can run evaluate() with GroundednessEvaluator with conversation-style input

Without this code change, running evaluate() with GroundednessEvaluator without `query` input, like in #1, would fail.

Reply:
Added a unittest to cover this explicitly